src/js/helpers/useToast.test.ts

```diff
@@ -0,0 +1,122 @@
+} from '@constants/mocks/useToast-data';
+
+describe('useToast', () => {
+  const expectedOptions = {
+    getType(hasType: boolean) {
+
+      return hasType ? `.bg-${testOptions.type}` : `.bg-${defaultType}`;
+    },
+    getClassList() {
+      const customClassList = testOptions.classlist;
+      let expectedClassList = '';
+
+      if (customClassList) {
+        customClassList.trim().split(' ').forEach((value) => {
+          
+          if (value) {
+            expectedClassList = expectedClassList.concat(' ', value);
+          }
+        });
+
+        return expectedClassList.trim();
+      }
+    },
+    getBtnClose() {
+      if (testOptions.autohide !== undefined && testOptions.autohide === false) {
+
+        return `.${btnCloseComponent}`;
+      }
+    }
+  }
+
+  const getToastElement = (hasType = false) => {
+    const toastContainer = document.querySelector<HTMLElement>(selectorsMap.toast.container);
+    const toastElement = toastContainer?.querySelector<HTMLElement>(`${selectorsMap.toast.toast}${expectedOptions.getType(hasType)}`);
+
+    return toastElement ?? null;
+  }
+
+  const getToastMessage = (toastElement: HTMLElement | null) => {
+    const toastBody = toastElement?.querySelector<HTMLElement>(`${selectorsMap.toast.toast}-body`);
+    
+    return toastBody?.innerHTML ?? '';
+  }
+
+  const isFallbackConnected = (document: Document) => {
+    const toastContainer = document.querySelector<HTMLElement>(selectorsMap.toast.container);
+
+    return toastContainer?.classList.contains(`${containerComponent}--fallback`);
+  }  
+
+  describe('with container and toast template existing in the DOM', () => {
+    beforeEach(() => {
+      const body = document.querySelector<HTMLBodyElement>('body');
+      
+      if (body) {
+        body.innerHTML = validTemplate;
+      }
+    });
+
+    it('should display the message', () => {
+      useToast(testMessage).show();
+      const toastElement = getToastElement();
+
+      expect(getToastMessage(toastElement) === testMessage).toBe(true);
+    });
+
+    it('should add custom class list to the element if exists', () => {
+      useToast(testMessage, testOptions).show();
+      const toastElement = getToastElement(true);
+      const expectedClassList = expectedOptions.getClassList();
+
+      if (expectedClassList) {
+        const pattern = new RegExp(expectedClassList);
+
+        expect(toastElement?.classList.toString().match(pattern)?.length).toBe(1);
+      }   
+    });
+
+    it('should display the close button if autohide is false', () => {
+      useToast(testMessage, testOptions).show();
+      const toastElement = getToastElement(true);
+      const expectedBtnClose = expectedOptions.getBtnClose();
+
+      if (expectedBtnClose) {
+        const btnCloseElement = toastElement?.querySelector<HTMLButtonElement>(expectedBtnClose);
+
+        expect(btnCloseElement?.classList.contains('d-none')).toBe(false);
+      }
```

Comment:
I don't understand much the use case here, so I don't know if you want that for the test : 
if `expectedBtnClose` is null, the test will pass (no assertion before). So if `expectedBtnClose` shouldn't be null, you can add an expect before and do your expect without a condition.

Reply:
`const testAutoHideOptions = { type: 'danger', autohide: false };`

![Screenshot from 2022-02-21 12-54-14](https://user-images.githubusercontent.com/85633460/154926009-178b1488-9f82-4aa8-9d72-88d5f57c4c27.png)

Wdyt?

Reply:
Looks good to me.

Reply:
Ty @boubkerbribri 

Your comment helped me to learn what's the purpose of the tests!